### ISSUE TITLE
Refresh Function

### ISSUE THREAD

**Issue by influxweb** [goldfire/miniTip#2]
First off, this is a fantastic plugin...lightweight and straight to the point.

I am trying to use it with dynamic content and am having an issue with re-initializing it. I have some color swatches that are re-drawn whenever one of them is clicked (this is due to server-side functions to check availability of the chosen swatch). On initial load, the tool tips show as expected. However, after the re-draw, the tool tip function no longer is associated with the swatches. Can a reload or refresh option be added into the plugin so it can deal with dynamic or AJAX driven content?

**Comment by goldfire** [comment]
Are you saying the content within the tooltip no longer shows, or the tooltip itself no longer shows?

**Comment by influxweb** [comment]
The tool tip itself doesn't show. I am basing this off the title tag holding the color swatch name. Although the miniTip tool tip doesn't show, the default browser behavior to show the contents of the title tag does work.

**Comment by goldfire** [comment]
Could you post the miniTip initialization code so I can get a better handle of what is going on?

**Comment by influxweb** [comment]
I ma using the minified version of the script and call it in this way: $('.tips').miniTip();

**Comment by goldfire** [comment]
Is it an option to just run $('.tips').miniTip() again, like in this jsbin?

http://jsbin.com/umalul/2/edit

**Comment by influxweb** [comment]
I have included the call to run it again in the function that generates the swatches, but it still doesn't fire.

```
AttributeMachine.prototype.Generate_Swatch = function (product_code, attribute, option) {
    var swatch = document.createElement('li');
    var span = document.createElement('span'); // to vertically center the swatch images
    var img = document.createElement('img');
    img.setAttribute('title', option.prompt);
    img.setAttribute('class', 'tips');
    img.src = option.image;
    swatch.appendChild(span);
    span.appendChild(img);

    return swatch;
    $('.tips').miniTip();
};
```

**Comment by goldfire** [comment]
Where is the swatch getting added to the DOM?

**Comment by influxweb** [comment]
This set of code is added to the page HEAD tag and populates a UL that is designated by an ID. I have tried calling the main miniTip.js file before this code and after the code with the same results.

**Comment by goldfire** [comment]
Well from looking at that code you posted, it appears $('.tips').miniTip(); is still getting called before the swatch gets added to the DOM.

For example, if you look at this modified jsbin I setup, it works when I add the swatch before calling miniTip, but not if you remove that line.

http://jsbin.com/umalul/4/edit#javascript,html,live

**Comment by influxweb** [comment]
Correct. On initial load of the page, everything works as expected. It is after the swatches are reloaded through AJAX via the "AttributeMachine.prototype.Generate_Swatch" function that they no longer work.

**Comment by goldfire** [comment]
Right, but the swatch doesn't get written to the DOM in that funciton (unless I'm seeing this wrong), so calling $('.tips').miniTip(); in that function won't work.

**Comment by influxweb** [comment]
Actually, that is the function that writes the swatches to the DOM on the reload. However, you got me thinking about how the initial swatches get loaded to the DOM. I was able to find an initialize function that I was able to add $('.tips').miniTip(); to and now it works on the reload.

Thank you for your help with this, I really appreciate it. keep up the great work, this has to be the best tool tip plugin that I have ever used.
